src/ManageCourses.Api/Services/UserLogService.cs

```diff
@@ -26,21 +28,24 @@ public bool CreateOrUpdateUserLog(string signInUserId, McUser user)
                         .SingleOrDefault(x => x.SignInUserId == signInUserId);
 
                     var add = userLog == null;
+                    var now = DateTime.UtcNow;
                     if (add)
                     {
                         userLog = new UserLog
                         {
-                            FirstLoginDateUtc = DateTime.UtcNow,
-                            SignInUserId = signInUserId
+                            FirstLoginDateUtc = now,
+                            SignInUserId = signInUserId,
+                            WelcomeEmailDateUtc = now
                         };
                     }
 
                     userLog.User = user;
                     userLog.UserEmail = user.Email;
-                    userLog.LastLoginDateUtc = DateTime.UtcNow;
+                    userLog.LastLoginDateUtc = now;
 
                     if (add)
                     {
+                        _welcomeEmailService.Send(user);
```

Comment:
It strikes me that the UserLog might be conceptually the wrong thing to be orchestrating the sending of the email. Think it breaks https://en.wikipedia.org/wiki/SOLID